src/index.ts

```diff
@@ -72,8 +72,24 @@ const prepareRun = async (
     const { value: testsResult } = await throwOnApiError(
       testrailAPI.getTests(existingRun.id)
     );
-    const currentCaseIds =
-      testsResult?.tests?.map((test) => test.case_id) || [];
+    let currentCaseIds = testsResult?.tests?.map((test) => test.case_id) || [];
+    let missingRes = false;
```

Comment:
You are repeating the same logic in 3 places, which in my opinion could be easily extracted to some helper function.
Would you be able to do that?
If you need some guidance, feel free to ask.

Reply:
Hi Damian, thanks for your answer. I try to improve my code using helper functions. Let me know if it is not yet correct.